src/goto-symex/symex_assign.cpp

```diff
@@ -231,10 +232,108 @@ void goto_symext::symex_assign_symbol(
     tmp_ssa_rhs.swap(ssa_rhs);
   }
 
+  // TODO: done via rename
+  // field_sensitivityt::apply(ns, ssa_rhs, false);
+
   state.rename(ssa_rhs, ns);
+
+  ssa_exprt lhs_mod = lhs;
+
+  if(
+    ssa_rhs.id() == ID_byte_update_little_endian ||
+    ssa_rhs.id() == ID_byte_update_big_endian)
+  {
+    byte_update_exprt &bu = to_byte_update_expr(ssa_rhs);
+    exprt be = byte_extract_exprt(
+      bu.id() == ID_byte_update_big_endian ? ID_byte_extract_big_endian
+                                           : ID_byte_extract_little_endian,
+      lhs_mod,
+      bu.offset(),
+      bu.value().type());
+    do_simplify(be);
+
+    if(be.id() == ID_symbol)
+    {
+      ssa_rhs = bu.value();
+      lhs_mod = to_ssa_expr(be);
+    }
+    else if(be.id() == ID_index)
+    {
+      ssa_rhs = bu.value();
+
+      while(be.id() == ID_index)
+      {
+        index_exprt &idx = to_index_expr(be);
+
+#ifdef USE_UPDATE
+        update_exprt new_rhs(idx.array().type());
+        new_rhs.old() = idx.array();
+        new_rhs.designator().push_back(index_designatort(idx.index()));
+        new_rhs.new_value() = ssa_rhs;
+#else
+        exprt new_rhs(ID_with, idx.array().type());
+        new_rhs.copy_to_operands(idx.array(), idx.index(), ssa_rhs);
+#endif
+
+        ssa_rhs = new_rhs;
```

Comment:
`new_rhs` isn't used afterwards, so assign to `ssa_rhs` directly when it is constructed

Reply:
Doesn't seem possible because `ssa_rhs` is used in constructing `new_rhs`.